src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,51 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
 public class WorkWithFile {
+    public static final String FIRST_KEYWORD = "supply";
+    public static final String SECOND_KEYWORD = "buy";
+    public static final String RESULT_KEYWORD = "result";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        writeOutputFile(toFileName, getOutputString(fromFileName));
+    }
+
+    private String getOutputString(String fromFileName) {
+        StringBuilder outputString = new StringBuilder();
+        try (BufferedReader inputFileBuffRdr = new BufferedReader(new FileReader(fromFileName))) {
+            String inputLine;
+            int[] countedBalances = {0, 0};
+            while ((inputLine = inputFileBuffRdr.readLine()) != null) {
+                String[] splittedInputLine = inputLine.split(",");
+                if (splittedInputLine[0].equals(FIRST_KEYWORD)) {
+                    countedBalances[0] += Integer.parseInt(splittedInputLine[1]);
+                } else if (splittedInputLine[0].equals(SECOND_KEYWORD)) {
+                    countedBalances[1] += Integer.parseInt(splittedInputLine[1]);
+                }
```

Comment:
The logic here still assumes that any line not starting with 'supply' is a 'buy' entry. Consider adding a check to ensure that each line starts with a recognized keyword before processing it. This will prevent incorrect processing of malformed lines or other types of entries.